pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -478,7 +478,7 @@ def is_subrepo(directory):
                 if repodata == DIST_TREE_MAIN_REPO_PATH:
                     treeinfo["repositories"].update({repodata: None})
                     continue
-                name = f"{repodata}-{treeinfo['hash']}"
+                name = f"{repodata}-{treeinfo['hash']}-{repository_pk}"
```

Comment:
No good reason that I'm aware of.
I think we did not want to deduplicate sub repos because their management looked complicated.
But I do not see a reason why not to create a new repo version.
The only question is how would you generate the repo name? If there were any changes, a treeinfo hash would likely be changed.

Reply:
Thinking more about it, I'm not sure how to use subrepos. It seems that there should be some dist tree identifier in the name. I think we included it so subrepos are not reused when dist tree changes, to handle potential rollbacks better.
Imagine that you synced a distree with subrepos into repo version 1 of the main repo.
Now something changed in a sub repo, and distree is slightly different now, so a new dist tree content unit should be created and added to a repo version 2 of the main repo.
If for sub repo changes we will just create a new repo version and relate the same subrepo to the newly created distree content, whenever we roll back a main repo to repo version 1, we'll have no info or control which repo version each sub repo is supposed to be at, we'll use the latest and it will be wrong.
```

 ----------------                      ----------------
| repo version 1 |                    | repo version 2 |
 ----------------                      ----------------
  | dist tree 1 |                       | dist tree 2 |
   -------------                         -------------
                   \                 /
                       ----------                          
                      | sub repo |                       
                       ----------
                         | v1 |
                          ----                             
                         | v2 | 
                          ----                             
```